tradelogs/cmd/trade-logs-crawler/main.go

```diff
@@ -144,21 +145,27 @@ func getTradeLogs(c *cli.Context) error {
 		return err
 	}
 
-	err = influxStorage.SaveTradeLogs(tradeLogs)
-	if err != nil {
-		return err
-	}
-
 	// fetch eth usd rate
 	ethUSDRateFetcher, err := tokenrate.NewETHUSDRateFetcher(sugar, dbName, influxClient, coingecko.New())
```

Comment:
Should this be cache provider. Following this logic then it will always be calling to coingecko's API for ETH-USD rate.

Reply:
```suggestion
	ethUSDRateFetcher, err := tokenrate.NewETHUSDRateFetcher(sugar, dbName, influxClient, coingecko.New())
```
As @favadi said, the provider might change and it could be different about the precision (now is by day but it could be hour or minute or even second), so that we leave it to call one time per block.